applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_analysis.py

```diff
@@ -0,0 +1,133 @@
+
+    #### Internal functions ####
+    def _CreateSolver(self, external_model_part=None):
+        """ Create the Solver (and create and import the ModelPart if it is not passed from outside) """
+        if external_model_part != None:
+            # This is a temporary solution until the importing of the ModelPart
+            # is removed from the solver (needed e.g. for Optimization)
+            if (type(external_model_part) != KM.ModelPart):
+                raise Exception("Input is expected to be provided as a Kratos ModelPart object")
+            self.using_external_model_part = True
+        else:
+            self.using_external_model_part = False
+
+        ## Get echo level and parallel type
+        self.echo_level = self.ProjectParameters["problem_data"]["echo_level"].GetInt()
+        self.parallel_type = self.ProjectParameters["problem_data"]["parallel_type"].GetString()
+
+        # To avoid many prints # TODO leave this?
+        if (self.echo_level == 0):
+            KM.Logger.GetDefaultOutput().SetSeverity(KM.Logger.Severity.WARNING)
+
+        ## Import parallel modules if needed
+        if (self.parallel_type == "MPI"):
+            import KM.mpi as KratosMPI
+            import KM.MetisApplication as MetisApplication
+            import KM.TrilinosApplication as TrilinosApplication
+            self.is_printing_rank = (KratosMPI.mpi.rank == 0)
+        else:
+            self.is_printing_rank = True
+
+        ## Structure model part definition
+        if self.using_external_model_part:
+            self.main_model_part = external_model_part
+        else:
+            main_model_part_name = self.ProjectParameters["problem_data"]["model_part_name"].GetString()
+            self.main_model_part = KM.ModelPart(main_model_part_name)
+            self.main_model_part.ProcessInfo.SetValue(KM.DOMAIN_SIZE,
+                                                      self.ProjectParameters["problem_data"]["domain_size"].GetInt())
+
+        ## Solver construction
+        import python_solvers_wrapper_contact_structural
+        self.solver = python_solvers_wrapper_contact_structural.CreateSolver(self.main_model_part, self.ProjectParameters)
+
+        ## Adds the necessary variables to the model_part only if they don't exist
+        self.solver.AddVariables()
+
+        if not self.using_external_model_part:
+            ## Read the model - note that SetBufferSize is done here
+            self.solver.ReadModelPart() # TODO move to global instance
+
+    def _ExecuteInitialize(self):
+        """ Initializing the Analysis """
+
+        super()._ExecuteInitialize()
+
+        ## Processes construction
+        import process_factory
+        if (self.ProjectParameters.Has("contact_process_list") is True):
+            self.list_of_processes += process_factory.KratosProcessFactory(self.structure_model).ConstructListOfProcesses(self.ProjectParameters["contact_process_list"])
```

Comment:
I would do the following:
~~~py
if (self.ProjectParameters.Has("contact_process_list") is True):
    contact_process = process_factory.KratosProcessFactory(self.structure_model).ConstructListOfProcesses(self.ProjectParameters["contact_process_list"])
    contact_process.ExecuteInitialize()
    self.list_of_processes += contact_process
~~~

Then you can remove the `self.list_of_processes[-1].ExecuteInitialize()` (which btw is wrong in case you dont have `contact_process_list` in the ProjectParams)

Reply:
True, I was thinking like I always have contact